src/handlers.ts

```diff
@@ -30,14 +54,42 @@ Please add me to an unencrypted chat.`);
       }
       await client.sendReadReceipt(roomId, event.event_id);
       await client.setTyping(roomId, true, 10000)
-      try {
-        // timeout after 2 minutes (which will also abort the underlying HTTP request)
-        const result = await chatGPT.sendMessage(question, {
-          timeoutMs: 2 * 60 * 1000
+
+      console.dir(event)
+      // proper threading rewrite
+      let chatGPTconversation
+      if (event.content["m.relates_to"] !== undefined) {
+        let storedValue = (client.storageProvider.readValue(event.content["m.relates_to"].event_id) as string)
+        if (storedValue !== undefined){
+          chatGPTconversation = storedValue.split(',')
+        } else {
+          console.log("Something went wrong with stored value, we couldn't split it ok.")
+        }
+      }
+
+      let storageKey = event.event_id // store event_id of the received message, it will be the message root if we just started a new thread.
+      console.log('Attempting to save event_id')
+      if (event.content["m.relates_to"] !== undefined) {
+        console.log('Detected existing relates_to saving root_event_id instead.')
+        storageKey = event.content["m.relates_to"].event_id // it is possible this thread already existed so we might need an older root, check for one.
+      }
+  
+      let result
```

Comment:
nit: could we add types to variable initialisations?

Reply:
Fixed.